roles/smartgateway/tasks/events.yml

```diff
@@ -0,0 +1,72 @@
+---
+# tasks file for smartgateway
+- name: Write out configmap for smart gateway
+  template:
+    dest: /tmp/configmap.yaml
+    src: "{{ service_type }}-configmap.yaml.j2"
+
+- name: Smart gateway ConfigMap
+  k8s:
+    src: /tmp/configmap.yaml
+
+- name: Start the smart gateway
+  k8s:
+    resource_definition:
+      apiVersion: apps.openshift.io/v1
+      kind: DeploymentConfig
+      metadata:
+        name: '{{ meta.name }}-smartgateway'
+        namespace: '{{ meta.namespace }}'
+      spec:
+        replicas: '{{ size }}'
+        template:
+          metadata:
+            labels:
+              app: smartgateway_events_consumer
+              managed-by: smartgateway-operator
+              smartgateway-name: "{{ meta.name }}"
+          spec:
+            securityContext:
+              nonroot: true
+            containers:
+            - name: smart-gateway
+              image: '{{ container_image_path }}'
+              volumeMounts:
+              - name: sg-config
+                mountPath: /config
+              args:
+                - '-config=/config/smartgateway_config.json'
+                - '-uname=$(MY_POD_NAME)'
+                - '-servicetype={{ service_type }}'
+              env:
+                - name: MY_POD_NAME
+                  valueFrom:
+                    fieldRef:
+                      apiVersion: v1
+                      fieldPath: metadata.name
+              ports:
+              - name: smart-gateway
+                containerPort: 8081
+            volumes:
+            - name: sg-config
+              configMap:
+                name: "{{ meta.name }}-smartgateway"
+            nodeSelector:
+              application: sa-telemetry
+
+- name: Service for the smart gateway
+  k8s:
+    resource_definition:
+      apiVersion: v1
+      kind: Service
```

Comment:
Maybe this is a question more than a comment, but how does a Service work without a selector? I would expect that this would create a service with no registered backends (I haven't tried it live yet)

Reply:
It's possible that just needs to be removed. I will investigate.